ulib/c_libax/src/network.c

```diff
@@ -0,0 +1,211 @@
+#include <ctype.h>
+#include <errno.h>
+#include <libax.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+
+#if defined(AX_CONFIG_ALLOC) && defined(AX_CONFIG_NET)
+/*Only IPv4. Ports are always 0. Ignore service and hint. Results' ai_flags, ai_socktype,
+ * ai_protocol and ai_canonname are 0 or NULL.  */
+int getaddrinfo(const char *__restrict node, const char *__restrict service,
+                const struct addrinfo *__restrict hints, struct addrinfo **__restrict res)
+{
+    struct sockaddr *addrs = (struct sockaddr *)malloc(MAXADDRS * sizeof(struct sockaddr));
+    int res_len = ax_resolve_socket_addr(node, addrs, MAXADDRS);
+    if (res_len < 0)
+        return EAI_FAIL;
+    if (res_len == 0)
+        return EAI_NONAME;
+    struct addrinfo *_res = (struct addrinfo *)calloc(res_len, sizeof(struct addrinfo));
+    for (int i = 0; i < res_len; i++) {
+        (_res + i)->ai_family = AF_INET;
+        (_res + i)->ai_addrlen = sizeof(struct sockaddr);
+        (_res + i)->ai_addr = (addrs + i);
+        (_res + i)->ai_next = (_res + i + 1);
```

Comment:
This causes bug in Redis:
`ai_socktype` and `ai_protocol` fields are not set. Did not support code like this: 
```
    if ((rv = getaddrinfo(bindaddr,_port,&hints,&servinfo)) != 0) {
        anetSetError(err, "%s", gai_strerror(rv));
        return ANET_ERR;
    }
    for (p = servinfo; p != NULL; p = p->ai_next) {
        if ((s = socket(p->ai_family,p->ai_socktype,p->ai_protocol)) == -1)
            continue;
        .........}
```